src/components/autocomplete/js/autocompleteController.js

```diff
@@ -327,9 +334,8 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
 
         // cancel results if search text is not long enough
         if (!isMinLengthMet()) {
-          ctrl.loading = false;
           ctrl.matches = [];
-          ctrl.hidden  = shouldHide();
+          setLoading(false);
```

Comment:
The approach I suggested should function exactly the same as the setter method, but you'd remove the part of that method when it actually sets the value.  This was a fairly minor complaint, so I'm fine leaving this.

It also shouldn't wait for a digest cycle using the proposed method.